pkg/services/correlations/api.go

```diff
@@ -0,0 +1,48 @@
+package correlations
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/grafana/grafana/pkg/api/response"
+	"github.com/grafana/grafana/pkg/api/routing"
+	"github.com/grafana/grafana/pkg/models"
+	ac "github.com/grafana/grafana/pkg/services/accesscontrol"
+	"github.com/grafana/grafana/pkg/services/datasources"
+
+	"github.com/grafana/grafana/pkg/web"
+)
+
+func (s *CorrelationsService) registerAPIEndpoints() {
+	uidScope := datasources.ScopeProvider.GetResourceScopeUID(ac.Parameter(":uid"))
+	authorize := ac.Middleware(s.AccessControl)
+
+	s.RouteRegister.Group("/api/datasources/uid/:uid/correlations", func(entities routing.RouteRegister) {
+		entities.Post("/", authorize(ac.ReqOrgAdmin, ac.EvalPermission(datasources.ActionWrite, uidScope)), routing.Wrap(s.createHandler))
```

Comment:
should this be `ReqOrgAdminOrEditor` instead?
```suggestion
		entities.Post("/", authorize(ac.ReqOrgAdminOrEditor, ac.EvalPermission(datasources.ActionWrite, uidScope)), routing.Wrap(s.createHandler))
```

Reply:
Nope, reusing the same logic of datasources.